### PR TITLE
network: static route - Allow to configure per route hop limit

### PR DESCRIPTION
**Pull request by ssahani**
```
[Route]
Gateway=172.16.130.2
HopLimit=34

```

```
❯ ip r
default via 172.16.130.2 dev ens33 proto static onlink hoplimit 34 
```

### CODE REVIEW

**Comment by bluca** [comment]
Can you add a test for this in one of the test/test-network/conf/25-route* files?

**Review by bluca — APPROVED** [review]

File: man/systemd.network.xml [hunk 1]
```diff
@@ -1617,6 +1617,13 @@ allow my_server_t localnet_peer_t:peer recv;</programlisting>
         </listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>HopLimit=</varname></term>
+        <listitem>
+          <para> Configures per route hop limit. See also <varname>IPv6HopLimit=</varname>.</para>
```
Comment (yuwata) [inline man/systemd.network.xml:1623]: Not a big matter, but unnecessary space after `<para>`.

Missing acceptable value range.

File: src/network/networkd-route.c [hunk 2]
```diff
@@ -2532,6 +2532,67 @@ int config_parse_route_type(
...
+        if (isempty(rvalue)) {
+                n->hop_limit = 0;
+                TAKE_PTR(n);
+                return 0;
+        }
+
+        r = safe_atou32(rvalue, &k);
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Could not parse route hop limit %s \"%s\", ignoring assignment: %m", lvalue, rvalue);
```
Comment (yuwata) [inline src/network/networkd-route.c:2576]: `parse route hop limit %s \"%s\"` -> `parse per route hop limit \"%s\"`

File: src/network/networkd-route.c [hunk 3]
```diff
@@ -2532,6 +2532,67 @@ int config_parse_route_type(
...
+                return 0;
+        }
+        if (k > 255) {
+                log_syntax(unit, LOG_WARNING, filename, line, 0,
+                           "Specified route hop limit %s \"%s\" is too large, ignoring assignment: %m", lvalue, rvalue);
```
Comment (yuwata) [inline src/network/networkd-route.c:2581]: Also here.

File: src/network/networkd-route.c [hunk 4]
```diff
@@ -2532,6 +2532,67 @@ int config_parse_route_type(
...
+                return 0;
+        }
+        if (k == 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, 0,
+                           "Invalid route hop limit %s \"%s\", ignoring assignment: %m", lvalue, rvalue);
```
Comment (yuwata) [inline src/network/networkd-route.c:2586]: Also here.

**Pull request merged**